crates/runner-nordic/src/storage.rs

```diff
@@ -90,20 +93,107 @@ impl Storage {
         unsafe { &*self.ptr }
     }
 
+    pub fn ptr(&self) -> *const u8 {
+        self.ptr as *const u8
+    }
+
     pub fn len(&self) -> usize {
         self.ptr.len()
     }
+}
 
-    pub const ERASE_SIZE: usize = PAGE_SIZE;
-    pub fn erase(&self, offset: usize, length: usize) -> StorageResult<()> {
-        let mut helper = Helper::new(self);
-        helper.nvmc().erase(offset as u32, (offset + length) as u32).map_err(convert)
+pub struct StorageWriter {
+    storage: Storage,
+    // None if not running.
+    dry_run: Option<bool>,
+    // offset + buffer.len() <= storage.len() && offset % WORD_SIZE == 0
```

Comment:
We never check and use `offset + buffer.len() <= storage.len()`, we just rely on the NVMC implementation. We can just remove this part of the comment.